src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -15,21 +15,35 @@ public override BoundNode VisitRangeExpression(BoundRangeExpression node)
             Debug.Assert(node != null && node.MethodOpt != null);
 
             bool needLifting = false;
-            var operandsBuilder = new ArrayBuilder<BoundExpression>();
+            var F = _factory;
 
             var left = node.LeftOperand;
             if (left != null)
             {
-                operandsBuilder.Add(tryOptimizeOperand(left));
+                left = tryOptimizeOperand(left);
+            }
+            else
+            {
+                // new Index(0, fromEnd: false)
+                left = F.New(
+                    WellKnownMember.System_Index__ctor,
+                    ImmutableArray.Create<BoundExpression>(F.Literal(0), F.Literal(false)));
             }
```

Comment:
Perhaps factor this block out: `left = optimizeOrCreateIndex(left, isLeft: true);`? Possibly could be factored into a single local function (ie. merge the logic with `tryOptimizeOperand`).